Capri.Services/Proposals/ProposalUpdater.cs

```diff
@@ -59,12 +73,42 @@ await _context
             }
 
             proposal = _mapper.Map(inputData, proposal);
+            AssignProposalStatus(proposal);
 
             _context.Proposals.Update(proposal);
             await _context.SaveChangesAsync();
 
             var proposalViewModels = _mapper.Map<ProposalViewModel>(proposal);
             return ServiceResult<ProposalViewModel>.Success(proposalViewModels);
         }
+
+        private bool NumOfStudentsExceedsTheMaximum(ICollection<Guid> students, int maxNumOfStudents)
```

Comment:
You can move that method to another service, because I've seen you use the same in ProposalGetter

Reply:
Done